### ISSUE TITLE
assertion failure from ud ('ooo_type == UCS_FRAG_LIST_INSERT_DUP' failed)

### ISSUE THREAD

**Issue by alinask** [openucx/ucx#972]
To reproduce:

/hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/installs/4Ftt/install/bin/mpirun -np 128 -mca btl_openib_warn_default_gid_prefix 0 --bind-to core --tag-output --timestamp-output --display-map -mca pml ucx -x UCX_SHM_DEVICES=all -x UCX_NET_DEVICES=mlx5_0:1 -x UCX_ACC_DEVICES=all -mca coll_hcoll_enable 0 -x UCX_TLS=rc_x,mm --map-by node /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/installs/4Ftt/tests/mpich_tests/mpich-mellanox.git/test/mpi/basic/simple

```
[vegas33:4576 :0]       ud_ep.c:495  Assertion `ooo_type == UCS_FRAG_LIST_INSERT_DUP' failed: OOO unsupported
[vegas34:4696 :0]       ud_ep.c:495  Assertion `ooo_type == UCS_FRAG_LIST_INSERT_DUP' failed: OOO unsupported
[vegas34:4695 :0]       ud_ep.c:495  Assertion `ooo_type == UCS_FRAG_LIST_INSERT_DUP' failed: OOO unsupported

/hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/uct/ib/ud/base/ud_ep.c: [ uct_ud_ep_rx_ctl() ]
      ...
      492     ooo_type = ucs_frag_list_insert(&ep->rx.ooo_pkts, &skb->u.ooo.elem, neth->psn);
      493     if (ooo_type != UCS_FRAG_LIST_INSERT_FAST) {
      494         ucs_assertv(ooo_type == UCS_FRAG_LIST_INSERT_DUP, "OOO unsupported");
==>   495         return;
      496     }
      497
      498     ep->dest_ep_id = ctl->conn_rep.src_ep_id;

[vegas08:4897 :0]       ud_ep.c:495  Assertion `ooo_type == UCS_FRAG_LIST_INSERT_DUP' failed: OOO unsupported
[vegas08:4898 :0]       ud_ep.c:495  Assertion `ooo_type == UCS_FRAG_LIST_INSERT_DUP' failed: OOO unsupported
==== backtrace ====
 0 0x000000000004b4be uct_ud_ep_rx_ctl()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/uct/ib/ud/base/ud_ep.c:495
 1 0x000000000004b4be uct_ud_ep_process_rx()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/uct/ib/ud/base/ud_ep.c:593
 2 0x0000000000051f8e uct_ud_mlx5_iface_poll_rx()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/uct/ib/ud/accel/ud_mlx5.c:395
 3 0x0000000000051f8e uct_ud_mlx5_iface_async_progress()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/uct/ib/ud/accel/ud_mlx5.c:453
 4 0x0000000000046029 uct_ud_iface_async_progress()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/uct/ib/ud/base/ud_iface.c:714
 5 0x0000000000046029 uct_ud_iface_timer()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/uct/ib/ud/base/ud_iface.c:750
 6 0x000000000003b421 ucs_async_dispatch_handler_cb()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/ucs/async/async.c:90
 7 0x000000000003c416 ucs_hashed_ucs_async_handler_t_find()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/ucs/async/async.c:23
 8 0x000000000003c52b ucs_async_dispatch_handler()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/ucs/async/async.c:126
 9 0x000000000003d0ba __ucs_async_poll_missed()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/ucs/async/async.c:375
10 0x000000000000cd6a ucs_async_check_miss()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/ucs/async/async.h:136
11 0x000000000000cd6a ucp_worker_progress()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/src/ucp/core/ucp_worker.c:434
12 0x0000000000003d13 mca_pml_ucx_progress()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/mpi-install/54Xi/src/ompi-release/ompi/mca/pml/ucx/pml_ucx.c:285
13 0x000000000002f872 opal_progress()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/mpi-install/54Xi/src/ompi-release/opal/runtime/opal_progress.c:187
14 0x000000000005a0d0 ompi_mpi_init()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/mpi-install/54Xi/src/ompi-release/ompi/runtime/ompi_mpi_init.c:825
15 0x000000000009e1fa PMPI_Init()  /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/mpi-install/54Xi/src/ompi-release/ompi/mpi/c/profile/pinit.c:84
16 0x00000000004024a2 main()  ??:0
17 0x0000000000021b15 __libc_start_main()  ??:0
18 0x00000000004023b9 _start()  ??:0
===================
[vegas33:04576] *** Process received signal ***
[vegas33:04576] Signal: Aborted (6)
[vegas33:04576] Signal code:  (-6)
[vegas33:04576] [ 0] /usr/lib64/libpthread.so.0(+0xf100)[0x7ffff7825100]
[vegas33:04576] [ 1] /usr/lib64/libc.so.6(gsignal+0x37)[0x7ffff74895f7]
[vegas33:04576] [ 2] /usr/lib64/libc.so.6(abort+0x148)[0x7ffff748ace8]
[vegas33:04576] [ 3] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libucs.so.2(__ucs_abort+0x10e)[0x7ffff0fd52fe]
[vegas33:04576] [ 4] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libuct.so.2(uct_ud_ep_process_rx+0xcae)[0x7ffff151e4be]
[vegas33:04576] [ 5] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libuct.so.2(+0x51f8e)[0x7ffff1524f8e]
[vegas33:04576] [ 6] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libuct.so.2(+0x46029)[0x7ffff1519029]
[vegas33:04576] [ 7] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libucs.so.2(+0x3b421)[0x7ffff0fc5421]
[vegas33:04576] [ 8] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libucs.so.2(ucs_hashed_ucs_async_handler_t_find+0x36)[0x7ffff0fc6416]
[vegas33:04576] [ 9] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libucs.so.2(ucs_async_dispatch_handler+0x4b)[0x7ffff0fc652b]
[vegas33:04576] [10] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libucs.so.2(__ucs_async_poll_missed+0x7a)[0x7ffff0fc70ba]
[vegas33:04576] [11] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/ucx_src/install/lib/libucp.so.2(ucp_worker_progress+0xaa)[0x7ffff175fd6a]
[vegas33:04576] [12] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/installs/4Ftt/install/lib/openmpi/mca_pml_ucx.so(mca_pml_ucx_progress+0x3a)[0x7ffff197dd13]
[vegas33:04576] [13] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/installs/4Ftt/install/lib/libopen-pal.so.13(opal_progress+0xa9)[0x7ffff6e9c872]
[vegas33:04576] [14] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/installs/4Ftt/install/lib/libmpi.so.12(ompi_mpi_init+0xdf9)[0x7ffff7a8c0d0]
[vegas33:04576] [15] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/installs/4Ftt/install/lib/libmpi.so.12(MPI_Init+0x180)[0x7ffff7ad01fa]
[vegas33:04576] [16] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/installs/4Ftt/tests/mpich_tests/mpich-mellanox.git/test/mpi/basic/simple[0x4024a2]
[vegas33:04576] [17] /usr/lib64/libc.so.6(__libc_start_main+0xf5)[0x7ffff7475b15]
[vegas33:04576] [18] /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/installs/4Ftt/tests/mpich_tests/mpich-mellanox.git/test/mpi/basic/simple[0x4023b9]
[vegas33:04576] *** End of error message ***
```

Currently doesn't reproduce on a smaller scale.
http://hpcweb.lab.mtl.com//hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20160913_063018_3018_100982_vegas06/html/test_stdout_MRl8WN.txt

It's a simple test:
```
$cat simple.c
/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
/*
 *  (C) 2001 by Argonne National Laboratory.
 *      See COPYRIGHT in top-level directory.
 */
#include "mpi.h"

int main(int argc, char *argv[])
{
    MPI_Init(&argc, &argv);
    MPI_Finalize();

    return 0;
}
```

**Comment by yosefe** [comment]
introduced by #963

**Comment by yosefe** [comment]
fixed by #972
